### DISCUSSION TITLE
Support groupby for multiple coordinates along the same dimension

### DISCUSSION THREAD

**Discussion by raj-magesh** [pydata/xarray#6044] [Ideas] [answered]
[As discussed here](https://stackoverflow.com/questions/60380993/groupby-multiple-coords-along-a-single-dimension-in-xarray), it would be great to have native support for grouping along multiple coordinates along the same dimension.

**Comment by dcherian** [comment, marked as answer]
``` python
import xarray as xr

d = xr.DataArray([[1, 2, 3], [4, 5, 6], [7, 8, 9], [10, 11, 12]],
    coords={
        'a': ('dim1',['A', 'A', 'B', 'B']),
        'b': ('dim1',['1', '2', '1', '2']),
        'c': ('dim2',['x', 'y', 'z'])
    },
    dims=['dim1', 'dim2'])
d
```
```
<xarray.DataArray (dim1: 4, dim2: 3)>
array([[ 1,  2,  3],
       [ 4,  5,  6],
       [ 7,  8,  9],
       [10, 11, 12]])
Coordinates:
    a        (dim1) <U1 'A' 'A' 'B' 'B'
    b        (dim1) <U1 '1' '2' '1' '2'
    c        (dim2) <U1 'x' 'y' 'z'
Dimensions without coordinates: dim1, dim2
```

Does [flox](https://flox.readthedocs.io/en/latest/generated/flox.xarray.xarray_reduce.html) do what you want?
``` python
import flox.xarray
flox.xarray.xarray_reduce(d, "a", "b", func="sum")
```
```
<xarray.DataArray (dim2: 3, a: 2, b: 2)>
array([[[ 1,  4],
        [ 7, 10]],

       [[ 2,  5],
        [ 8, 11]],

       [[ 3,  6],
        [ 9, 12]]])
Coordinates:
  * a        (a) object 'A' 'B'
  * b        (b) object '1' '2'
Dimensions without coordinates: dim2
```

**Reply by raj-magesh** [reply]
Yes, it does! It'd be great when it gets integrated into xarray.

**Reply by rsignell-usgs** [reply]
This is now integrated, right?   
``` python
import flox.xarray
flox.xarray.xarray_reduce(d, "a", "b", func="sum")
``` 
What would the syntax for this be in xarray?

**Reply by keewis** [reply]
as far as I can tell that's still up for debate: #6610 proposed some changes which are partially implemented in #6689

**Reply by dcherian** [reply]
Yes, the next big lift is to refactor the constructor, then enable all kinds of new things. 

For now, using `flox.xarray.xarray_reduce` (or `xhistogram`) is the way.
